### PR TITLE
Allow running as console application on Windows

### PR DESCRIPTION
**Pull request by MondayHopscotch**
### Required for all PRs:

- [X] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [X] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [N/A] README.md updated (if adding a new plugin)

Allow windows service to run as console application

### CODE REVIEW

**Comment by aMaZing0x41** [comment]
:+1: This is useful when one does not necessarily want to run telegraf as a stand-alone native service. It will essentially allow one to bypass telegraf running as a service so that it can be invoked by external applications. This change was primarily introduced in PR #1543.

**Comment by danielnelson** [comment]
It seems to me that the absence of the `-service` flag should cause Telegraf to run in the foreground, and if Telegraf is in the foreground there should be a console window.

**Comment by aMaZing0x41** [comment]
@danielnelson I don't believe that the `-service ` flag is in play here. Currently telegraf will _always_ attempt to run as a service when using a non-interactive login. There is no way to override this behavior. 
If one uses an interactive login telegraf seems to start up and run as console app.  In this specific case, we are using a wrapper exe, the login then gets passed to telegraf. Because the login is non-interactive telegraf is attempting to run as a service and failing. Does that make sense?

**Comment by danielnelson** [comment]
Got it.  What do you think about the flag being `-console`?  Also, can you make sure to test the signal handling as the reloadLoop branch was previously unix only. `https://golang.org/pkg/os/signal/#hdr-Windows`.

**Comment by aMaZing0x41** [comment]
I think the `-console` flag makes sense. We will take a look at the signal handling. Hopefully within the next week or so. Thanks for the feedback!

**Comment by aMaZing0x41** [comment]
I have tested these changes on a recent version of master (5bac08662e1376918ea3f70155cec8b4e95a0ffc). All appears well. @danielnelson mentioned the signaling. As far as I can see the exe is accepting signals. The process is stops cleanly. You mentioned the reloadLoop. Is there anything else to test regarding that?

File: cmd/telegraf/telegraf.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ var fUsage = flag.String("usage", "",
 	"print usage for a plugin, ie, 'telegraf -usage mysql'")
 var fService = flag.String("service", "",
 	"operate on the service")
+var fRunAsConsole = flag.Bool("run-as-console", false, "run as console application (windows only)")
 
 // Telegraf version, populated linker.
 //   ie, -ldflags "-X main.version=`git describe --always --tags`"
```
Comment (aMaZing0x41) [inline cmd/telegraf/telegraf.go]: Perhaps the usage string should be updated to contain the new `console` flag?

Comment (danielnelson) [reply cmd/telegraf/telegraf.go]: Yeah just rename the flag and we can merge.

**Comment by aMaZing0x41** [comment]
@danielnelson Thanks. @MondayHopscotch updated the flag to `console`. The circleci tests are failing, but it doesn't appear to be related to this change. Is this a known issue? Please let us know if we need to update anything else.

**Comment by danielnelson** [comment]
Tests would pass if the branch was merged/rebased, looks like the failures are what needed fixed when me move to Go 1.9.

**Pull request merged**